p22/benches/bench.rs

```diff
@@ -0,0 +1,163 @@
+#![feature(test)]
+extern crate test;
+
+#[cfg(test)]
+mod tests {
```

Comment:
You don't need `#[cfg(test)] mod tests { .. }` in benchmarks. We use it for unit test to reduce amount of compiled code when crates is used as a  library or binary.  But for benchmarks `#[cfg(test)]` is always true.